web/wp-content/plugins/tmbi-salesforce-dmp/class-tmbi-salesforce-dmp.php

```diff
@@ -0,0 +1,343 @@
+<?php
+/**
+TMBI Salesforce DMP
+
+@package TMBI Salesforce DMP
+Plugin Name: TMBI Salesforce DMP
+Version: 1.0.0
+Description: This plugin adds Salesforce DMP tags to the header and uses GPT to call Krux segments <a href='https://readersdigest.atlassian.net/browse/PLT-459' target='_blank'>Read more at PLT-459 ...</a>
+Author: DRDT Team
+Text Domain: tmbi-salesforce-dmp
+ */
+
+/**
+ * Including file for author details.
+ *
+ * @file
+ */
+require_once 'inc/author-detail.php';
+
+/**
+ *  Class TMBI_Salesforce_DMP.
+ */
+class TMBI_Salesforce_DMP {
+	const VERSION                = '1.0.0';
+	const DEPENDS                = null;
+	const PRIORITY               = 1;
+	const SCRIPT_VAR             = 'krux_data';
+	const SCRIPT_SLUG            = 'krux-click-tracking';
+	const SCRIPT_URL             = 'js/krux-click-traking.js';
+	const CONTROL_TAG_START      = '<!-- BEGIN Salesforce DMP ControlTag for "tmbi.com" -->';
+	const CONTROL_TAG_END        = '<!-- END Salesforce DMP ControlTag -->';
+	const CONTROL_SCRIPT_TAG     = "<script class='kxct' data-id='s9xpab5u5' data-timing='async' data-version='3.0' type='text/javascript'>%s</script>";
+	const INTERCHANGE_SCRIPT_TAG = "<script class='kxint' data-namespace='trustedmediabrandsinc' type='text/javascript'>%s</script>";
+
+	/**
+	 * Variable to exclude the taxonomies
+	 *
+	 * @var array
+	 */
+	public static $excluded_taxonomies = array( 'yst_prominent_words', 'post_tag', 'category', 'exclude_feed', 'post_format', 'author' );
+
+	/**
+	 * Variable to build the krux data for localization
+	 *
+	 * @var array
+	 */
+	public static $krux_data_layer = array();
+
+	/**
+	 * Initiate the methods
+	 */
+	public static function init() {
+		add_action( 'wp_head', array( __CLASS__, 'render_salesforce_dmp_tags' ), self::PRIORITY );
+		add_action( 'wp_enqueue_scripts', array( __CLASS__, 'load_krux_click_tracking' ) );
+		add_action( 'wp_footer', array( __CLASS__, 'remove_krux_tracking' ), 1 );
+	}
+
+	/**
+	 * Localize the Krux targeting data
+	 */
+	public static function load_krux_click_tracking() {
+		wp_register_script(
+			self::SCRIPT_SLUG,
+			plugin_dir_url( __FILE__ ) . self::SCRIPT_URL,
+			array( 'jquery' ),
+			self::VERSION,
+			false
+		);
+		if ( class_exists( 'Ad_Stack' ) ) {
+			wp_localize_script(
+				'ad-stack',
+				self::SCRIPT_VAR,
+				self::krux_build_datalayer()
+			);
+		}
+	}
+
+	/**
+	 * Add the salesforce tags
+	 */
+	public static function render_salesforce_dmp_tags() {
+
+		$control_tag_script = PHP_EOL . "window.Krux||((Krux=function(){Krux.q.push(arguments)}).q=[]);
+  (function(){
+    var k=document.createElement('script');k.type='text/javascript';k.async=true;
+    k.src=(location.protocol==='https:'?'https:':'http:')+'//cdn.krxd.net/controltag/s9xpab5u5.js';
```

Comment:
`s9xpab5u5` also appears as part as `CONTROL_SCRIPT_TAG`. Should this be moved to config, to support other sites?

Reply:
@fariasf as this is same for all our brands except for RDCA. I thought of keeping it here.